src/snapshot.rs

```diff
@@ -79,17 +80,20 @@ pub struct SnapshotCreator([usize; 1]);
 impl SnapshotCreator {
   /// Create and enter an isolate, and set it up for serialization.
   /// The isolate is created from scratch.
-  pub fn new(external_references: Option<&'static ExternalReferences>) -> Self {
+  pub fn new(
+    external_references: Option<&'static ExternalReferences>,
+    startup_data: Option<StartupData>,
+  ) -> Self {
     let mut snapshot_creator: MaybeUninit<Self> = MaybeUninit::uninit();
-    let external_references_ptr = if let Some(er) = external_references {
-      er.as_ptr()
-    } else {
-      std::ptr::null()
-    };
+    let external_references_ptr =
+      external_references.map_or_else(std::ptr::null, |er| er.as_ptr());
+    let startup_data_ptr = startup_data
+      .map_or_else(std::ptr::null_mut, |sd| Box::into_raw(Box::new(sd)));
     unsafe {
       v8__SnapshotCreator__CONSTRUCT(
         &mut snapshot_creator,
         external_references_ptr,
+        startup_data_ptr,
       );
       snapshot_creator.assume_init()
```

Comment:
It is not safe to return a stack-allocated SnapshotCreator. It causes the SnapshotCreator to potentially move -- which is incompatible with the C++ class definition. I think it would be best to bind `new` and `delete` in C++ rather than the constructor/destructor. Alternatively you could construct it into a box.

Reply:
wait, how does this have anything to do with my changes? I haven't touched the way the SnapshotCreator is constructed?

Reply:
Actually I take that back - SnapshotCreator is just a wrapper around a pointer to `struct SnapshotCreatorData`.